nautobot/users/views.py

```diff
@@ -250,30 +248,45 @@ class SavedViewUIViewSet(
     table_class = SavedViewTable
     action_buttons = ("export",)
 
-    def get_required_permission(self):
+    def get_queryset(self):
         """
-        Obtain the permissions needed to perform certain actions on a model.
+        Get the list of items for this view.
+        All users should be able to see saved views so we do not apply extra permissions.
         """
-        queryset = self.get_queryset()
-        try:
-            actions = [self.get_action()]
-            for i, item in enumerate(actions):
-                # Enforce users:change_savedview instead of users:clear_savedview
-                if item == "clear":
-                    actions[i] = "change"
-        except KeyError:
-            messages.error(
-                self.request,
-                "This action is not permitted. Please use the buttons at the bottom of the table for Bulk Delete and Bulk Update",
-            )
-        return self.get_permissions_for_model(queryset.model, actions)
+        return self.queryset
+
+    def check_permissions(self, request):
+        """
+        Check whether the user has the permissions needed to perform certain actions.
+        """
+        user = self.request.user
+        permission_required = self.get_required_permission()
+        # Check that the user has been granted the required permission(s) one by one.
+        # In case the permission has `message` or `code`` attribute, we want to include those information in the permission_denied error.
+        for permission in permission_required:
+            # If the user does not have the permission required, we raise DRF's `NotAuthenticated` or `PermissionDenied` exception
+            # which will be handled by self.handle_no_permission() in the UI appropriately in the dispatch() method
+            # Cast permission to a list since has_perms() takes a list type parameter.
+            if "savedview" in permission:
+                continue
+            if not user.has_perms([permission]):
+                self.permission_denied(
+                    request,
+                    message=getattr(permission, "message", None),
+                    code=getattr(permission, "code", None),
+                )
+
+    def list(self, request, *args, **kwargs):
+        if isinstance(request.user, AnonymousUser):
+            return self.handle_no_permission()
```

Comment:
Should be able to use `permission_classes = (IsAuthenticated,)` at the class level so as to not need to specify this in each action method.

Reply:
`permisssion_classes` does not seem to work. I will investigate more in the morning.

Reply:
No luck on this?

Reply:
OK yeah... we should be following or at least supporting the patterns used in DRF. Follow up issue is fine for now.